src/Api/Search.php

```diff
@@ -0,0 +1,299 @@
+     */
+    protected $query;
+
+    /**
+     * @var array
+     */
+    protected $sort;
+
+    /**
+     * @var int
+     */
+    protected $pageNumber;
+
+    /**
+     * @var int
+     */
+    protected $pageSize;
+
+    /**
+     * @var string
+     */
+    protected $highlightRegex;
+
+    /**
+     * @var int
+     */
+    protected $highlightLimit;
+
+    /**
+     * @var string
+     */
+    protected $sessionToken;
+
+    /**
+     * @inheritDoc
+     */
+    public function search(): SearchResult
+    {
+        /** @var SearchResult $search */
+        $search = $this->serializer->deserialize(
+            $this->send('POST')->getBody()->__toString(),
+            SearchResult::class,
+            'json'
+        );
+        return $search;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getConfigSchema(): array
+    {
+        return [
+            'apiKey' => [
+                'type' => 'string',
+                'required' => true,
+            ],
+            'searchApiEndpoint' => $this->getEndpointSchema(),
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getEndpointUri(): string
+    {
+        return $this->getConfigValue('searchApiEndpoint');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getRequestUriQuery(UriInterface $uri): array
+    {
+        $query = parent::getRequestUriQuery($uri);
+
+        $query['apiKey'] = $this->getConfigValue('apiKey');
+        $query['text'] = $this->getText();
+
+        if ($pageNumber = $this->getPageNumber()) {
+            $query['pageNumber'] = $pageNumber;
+        }
+        if ($pageSize = $this->getPageSize()) {
+            $query['pageSize'] = $pageSize;
+        }
+        if ($highlightRegex = $this->getHighlightRegex()) {
+            $query['highlightRegex'] = $highlightRegex;
+            if ($highlightLimit = $this->getHighlightLimit()) {
+                $query['highlightLimit'] = $highlightLimit;
+            }
+        }
+        if ($sessionToken = $this->getSessionToken()) {
+            $query['sessionToken'] = $sessionToken;
+        }
+
+        return $query;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getRequestMultipartStreamElements(): array
+    {
+        $parts = [];
+
+        if ($languages = $this->getLanguages()) {
+            $parts['languages'] = $this->jsonEncoder->encode($languages, 'json');
+        }
+        if ($query = $this->getQuery()) {
+            $parts['query'] = $this->jsonEncoder->encode($query, 'json');
+        }
+        if ($sort = $this->getSort()) {
+            $parts['sort'] = $this->jsonEncoder->encode($sort, 'json');
+        }
+
+        return $parts;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setText(?string $text): SearchInterface
+    {
+        $this->text = $text;
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getText(): ?string
```

Comment:
this will never be null as '***' is returned even if not set
```suggestion
    public function getText(): string
```